packages/editor/src/components/table-of-contents/index.js

```diff
@@ -26,7 +26,7 @@ function TableOfContents( { hasBlocks } ) {
 					aria-disabled={ ! hasBlocks }
 				/>
 			) }
-			renderContent={ () => <TableOfContentsPanel /> }
+			renderContent={ ( { onClose } ) => <TableOfContentsPanel onRequestClose={ onClose } /> }
```

Comment:
Was it ever discussed whether closing the TableOfContents upon making a selection is the intended behavior? It does not seem within the scope of what's meant to be addressed as part of #7142.

Reply:
Yes, this was discussed on this pr- https://github.com/WordPress/gutenberg/pull/10815#issuecomment-437581453

Reply:
Not closing the outline on click prevents a usability issue - can you remove if you prefer we address on a follow up PR.